runtime/compilers/rillv1/parse_chart.go

```diff
@@ -40,14 +38,24 @@ func (p *Parser) parseChart(node *Node) error {
 		return errors.New(`failed to parse "vega_lite" as JSON`)
 	}
 
-	if (tmp.Data.MetricsSQL == "" && tmp.Data.API == "") || (tmp.Data.MetricsSQL != "" && tmp.Data.API != "") {
-		return fmt.Errorf("exactly one of metrics_sql or api should be set")
+	if tmp.Data == nil {
+		return errors.New("data is mandatory")
+	}
+	resolver, resolverProps, err := p.parseDataYAML(tmp.Data)
+	if err != nil {
+		return err
 	}
 
 	if tmp.Data.API != "" {
 		node.Refs = append(node.Refs, ResourceName{Kind: ResourceKindAPI, Name: tmp.Data.API})
 	}
 
+	// Convert resolver properties to structpb.Struct before inserting the resource (since we can't return errors after that point)
+	resolverPropsPB, err := structpb.NewStruct(resolverProps)
+	if err != nil {
+		return fmt.Errorf("encountered invalid property type: %w", err)
+	}
```

Comment:
Actually, can we also move this into the `parseDataYAML` function, and return a `structpb` value instead of a map? Since otherwise, we will need to duplicate it every place that has a data section.